base/combinatorics.jl

```diff
@@ -276,7 +276,7 @@ julia> B[invperm(v)]
 """
 function invperm(a::AbstractVector)
     require_one_based_indexing(a)
-    b = zero(a) # similar vector of zeros
+    fill!(similar(a), zero(eltype(a))) # similar vector of zeros
```

Comment:
We're filling each entry in `b` anyways, so we don't actually need to initialize it with zeros.

```suggestion
    b = similar(a)
```

Reply:
This is necessary, as the zero check is carried out in there loop